pkg/apis/core/v1alpha1/defaults_test.go

```diff
@@ -444,6 +445,30 @@ var _ = Describe("Defaults", func() {
 			SetDefaults_Shoot(obj)
 			Expect(obj.Spec.Kubernetes.KubeAPIServer.EnableAnonymousAuthentication).To(PointTo(BeTrue()))
 		})
+
+		It("should default cri.name for k8s versions >=1.22 to containerd", func() {
+			obj.Spec.Kubernetes.Version = "1.22"
+			obj.Spec.Provider.Workers = []Worker{
+				{Name: "DefaultWorker"},
+				{Name: "Worker with CRI configuration",
+					CRI: &CRI{Name: CRIName("some configured value")}},
+			}
+			SetDefaults_Shoot(obj)
+			assertCRIName(obj.Spec.Provider.Workers[0], CRINameContainerD)
+			assertCRIName(obj.Spec.Provider.Workers[1], CRIName("some configured value"))
```

Comment:
Nit: I would vote for:

```suggestion
			Expect(obj.Spec.Provider.Workers[0].CRI).NotTo(BeNil())
			Expect(obj.Spec.Provider.Workers[0].CRI.Name).To(Equal(CRINameContainerD))
			Expect(obj.Spec.Provider.Workers[1].CRI).NotTo(BeNil())
			Expect(obj.Spec.Provider.Workers[1].CRI.Name).To(Equal(CRIName("some configured value")))
```

IMO having small assert functions is fine, but I really love when the assertion is done in the test (not in a helper func) (even accepting the fact that there might be a small code duplication) and you don't have to jump to the end of the file to check what is being asserted. Just a personal preference.

Reply:
I'm fine either way. Usually, I try to choose names which allow understanding what the `assert` function does without looking at it, but I absolutely understand your point of view here. In this instance the gain of separating out the assertments to a new function is probably debatable, so I'll change it.